src/sentry/queue/routers.py

```diff
@@ -0,0 +1,103 @@
+    """
+    Returns the split queue to use for a Celery queue.
+    Split queues allow us to spread the load of a queue to multiple ones.
+
+    This takes in input a queue name and returns the split. It is supposed
+    to be used by the code that schedules the task.
+
+    Each split queue can be individually rolled out via options.
+
+    WARNING: Do not forget to configure your workers to listen to the
+    queues appropriately before you start routing messages.
+    """
+
+    def __init__(self) -> None:
+        known_queues = _get_known_queues()
+        self.__queue_routers = {}
+        for source, destinations in settings.CELERY_SPLIT_QUEUE_ROUTES.items():
+            assert source in known_queues, f"Queue {source} in split queue config is not declared."
+            _validate_destiantions(destinations)
+            self.__queue_routers[source] = cycle(destinations)
+
+    def route_for_queue(self, queue: str) -> str:
+        rollout_rate = options.get("celery_split_queue_rollout").get(queue, 0.0)
+        if random.random() >= rollout_rate:
+            return queue
+
+        if queue in set(options.get("celery_split_queue_legacy_mode")):
+            # Use legacy route
+            # This router required to define the routing logic inside the
+            # settings file.
+            return settings.SENTRY_POST_PROCESS_QUEUE_SPLIT_ROUTER.get(queue, lambda: queue)()
+        else:
+            router = self.__queue_routers.get(queue)
+            if router is not None:
+                return next(router)
+            else:
+                return queue
+
+
+class TaskRoute(NamedTuple):
+    default_queue: str
+    queues: Iterator[str]
+
+
+class SplitQueueTaskRouter:
+    """
+    Routes tasks to split queues.
+
+    As for `SplitQueueRouter` this is meant to spread the load of a queue
+    to a number of split queues.
+
+    The main difference is that this is a router used directly by Celery.
+    It is configured as the main router via the `CELERY_ROUTES` setting.
+    Every time a task is scheduled that does not define a queue this router
+    is used and it maps a task to a queue.
+
+    Here as well, split queues can be rolled out individually via options.
+    """
+
+    def __init__(self) -> None:
+        known_queues = _get_known_queues()
+
+        self.__task_routers = {}
+        for source, destinations in settings.CELERY_SPLIT_QUEUE_TASK_ROUTES.items():
+            default_destination = destinations[1]
+            assert (
+                default_destination in known_queues
+            ), f"Queue {default_destination} in split queue config is not declared."
+            _validate_destiantions(destinations[0])
+
+            self.__task_routers[source] = TaskRoute(default_destination, cycle(destinations[0]))
+
+    def route_for_task(self, task: str, *args: Any, **kwargs: Any) -> Mapping[str, str] | None:
+        route = self.__task_routers.get(task)
+        if route is None:
+            return None
```

Comment:
@wedamija  I think this behavior is equivalent to not having a router. Does this seem right to you regarding the old split queue? As the existence of the router cannot be easily rolled out gradually (it is a setting)it would be good to be certain about this behavior. 
If it is unclear I can roll this entire behavior in s4s first by changing the config only there.

Reply:
I don't know this behaviour well, but from some reading online it seems that if `route_task` returns None then it will fall back to the default routing behaviour

Reply:
Ok, no pb. I will test the change in s4s first.

Reply:
From reading through celery's [Router](https://github.com/celery/celery/blob/118e004856e231f3c40f9badd75155d0dc0de38e/celery/app/routes.py#L66-L111) having a queue/route of none seems like it will be ok. I wasn't able to track down how the queue decision makes it way to amqp publishing but I would assume that the queue will not be replaced when routers don't make a decision.